### PR TITLE
fix: remove unexpected validate webhook in configuration

### LINKED ISSUE #2254: somethings different between charts and artifacts
1. a validate webhook config defined in:  

https://github.com/karmada-io/karmada/blob/master/charts/karmada/templates/_karmada_webhook_configuration.tpl#L62

while no such configuration in https://github.com/karmada-io/karmada/blob/master/artifacts/deploy/webhook-configuration.yaml

which one is correct？

### PR DESCRIPTION
**Pull request by zhixian82**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
cleanup misconfigured validation webhook from the chart.

**Which issue(s) this PR fixes**:
Fixes #2254

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
/assign @lonelyCZ 
Can you help to confirm it?

**Comment by zhixian82** [comment]
Can this pr and [pr](https://github.com/karmada-io/karmada/pull/2258) be merged into release 1.2?

File: charts/karmada/templates/_karmada_webhook_configuration.tpl [hunk 1]
```diff
@@ -51,20 +51,6 @@ webhooks:
...
     timeoutSeconds: 3
-  - name: clusteroverridepolicy.karmada.io
-    rules:
-      - operations: ["CREATE", "UPDATE"]
-        apiGroups: ["policy.karmada.io"]
-        apiVersions: ["*"]
-        resources: ["clusteroverridepolicies"]
-        scope: "Cluster"
-    clientConfig:
-      url: https://{{ $name }}.{{ $namespace }}.svc:443/validate-clusteroverridepolicy
```
Comment (lonelyCZ) [inline charts/karmada/templates/_karmada_webhook_configuration.tpl:62]: Why remove this? We have this webhook endpoint.

https://github.com/karmada-io/karmada/blob/e237614954a64b2595b9ec18128834a538975518/cmd/webhook/app/webhook.go#L122

Comment (zhixian82) [reply charts/karmada/templates/_karmada_webhook_configuration.tpl:62]: it's a validate admission, should moved to ValidatingWebhookConfiguration

Comment (lonelyCZ) [reply charts/karmada/templates/_karmada_webhook_configuration.tpl:62]: Great, nice finding!

**Comment by lonelyCZ** [comment]
Hi, @zhixian82 

Recommand to use following template to descripe this pr.

**What type of PR is this?**



**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Comment by zhixian82** [comment]
> Hi, @zhixian82
> 
> Recommand to use following template to descripe this pr.

done

**Comment by RainbowMango** [comment]
Thanks @lonelyCZ for pointing that. @zhixian82 might don't know we have the [PR template](https://github.com/karmada-io/karmada/blob/master/.github/PULL_REQUEST_TEMPLATE.md).

I've updated the PR description.  Thank you.
/approve

**Comment by RainbowMango** [comment]
@zhixian82 
Can you help to [cherry-pick](https://github.com/karmada-io/karmada/blob/master/docs/contributors/devel/cherry-picks.md) this patch to the following branches?:
- release-1.0
- release-1.1
- release-1.2

**Pull request merged**